ext/soap/soap.c

```diff
@@ -1307,19 +1307,14 @@ PHP_METHOD(SoapServer, handle)
 			sapi_add_header(hdr, sizeof("Location: ")+strlen(service->sdl->source)-1, 1);
 			efree(hdr);
 */
-			zval readfile, readfile_ret, param;
 
 			sapi_add_header("Content-Type: text/xml; charset=utf-8", sizeof("Content-Type: text/xml; charset=utf-8")-1, 1);
-			ZVAL_STRING(&param, service->sdl->source);
-			ZVAL_STRING(&readfile, "readfile");
-			if (call_user_function(EG(function_table), NULL, &readfile, &readfile_ret, 1, &param ) == FAILURE) {
-				soap_server_fault("Server", "Couldn't find WSDL", NULL, NULL, NULL);
```

Comment:
I realize this error became dead a long time ago, when the behaviour of `call_user_function` changed. However, the intention truly is to have this `soap_server_fault` call on failure, so please do that. Makes me think that we probably don't want REPORT_ERRORS then...

Reply:
Okay so if you don't get a stream, emit this soap error then.

Reply:
Yes